pkg/channel/message_dispatcher.go

```diff
@@ -105,29 +111,28 @@ func (d *MessageDispatcherImpl) DispatchMessageWithRetries(ctx context.Context,
 	// Otherwise, they are filled with the original message
 	var responseMessage cloudevents.Message
 	var responseAdditionalHeaders nethttp.Header
-
+	var dispatchExecutionInfo DispatchExecutionInfo
```

Comment:
This should be a pointer to `DispatchExecutionInfo`

Reply:
I agree I think it can help with stuff building on top of this like the eventing-kafka thing where the reporter could be option if we want.